src/chess.ts

```diff
@@ -1655,7 +1659,7 @@ export class Chess {
     let moves = ms.trim().split(new RegExp(/\s+/))
 
     /* delete empty entries */
-    moves = moves.join(',').replace(/,,+/g, ',').split(',')
+    moves = moves.filter(move => move !== '')
```

Comment:
The current method did not handle the edge case where `moves = ['']`:
```js
moves = [''].join(',').replace(/,,+/g, ',').split(',');
console.log(moves);
// output: ['']
```

The edge case happens when `ms` is empty (in the case where there are no moves):
```js
let ms = '';
console.log(ms.trim().split(new RegExp(/\s+/)));
// output: ['']
```

The solution I went with here is to just use the `.filter` method to remove any empty string from the array.